### DISCUSSION TITLE
communication client server in two direction

### DISCUSSION THREAD

**Discussion by anass-py** [contiki-ng/contiki-ng#1570] [Q&A]
hello 
we are using Contiki for our wireless communication projects.
the problem that we're now facing is : 
suppose we have 2 nodes ( A and B) , we want the  A send to B a data1 , and also B send to A another data2.
A can easily send to B and B receive it , however when we want to send it by the node B , A receive nothing.
should we make two different connection one for sending other for receiving ??
we'll appreciate any kind of help .

**Comment by atiselsts** [comment]
You need to elaborate the setup - what protocols are you using? What configuration? Does B have a route to A?

**Reply by anass-py** [reply]
we are begginer on contiki .
we are using tcp protocol, the probleme is we need a protocol for routing because we will have a mesh of relay nodes between A and B , so we need to choose a protocol of routing that allow us to communicate between A and B in both direction . 
if we use RPL as a protocol we need to configure B as a PRL_DAG and i don't assume that B will be able to send to A if it is configured as an RPL_DAG (final node destination) ?
have you an ideo about the communication in bothe direction between nodes with the use of a particular routing protocol ?
